pkg/v1/providers/ytt/02_addons/secretgen-controller/add_secretgen-controller.yaml

```diff
@@ -0,0 +1,24 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:template", "template")
+#@ load("@ytt:data", "data")
+#@ load("@ytt:yaml", "yaml")
+#@ load("/lib/helpers.star", "ValuesFormatStr")
+#@ load("secretgen-controller_addon_data.lib.yaml", "secretgencontrollerdatavalues")
+
+#@ if data.values.PROVIDER_TYPE != "tkg-service-vsphere" and data.values.SECRETGEN_CONTROLLER_ENABLE:
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: #@ "{}-secretgen-controller-addon".format(data.values.CLUSTER_NAME)
+  namespace: #@ data.values.NAMESPACE
+  labels:
+    tkg.tanzu.vmware.com/addon-name: "secretgen-controller"
+    tkg.tanzu.vmware.com/cluster-name: #@ data.values.CLUSTER_NAME
+    clusterctl.cluster.x-k8s.io/move: ""
```

Comment:
`clusterctl.cluster.x-k8s.io/move` should be added only for mgmt cluster

Reply:
We do in cpi and csi. See https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/providers/ytt/02_addons/csi/csi_secret.yaml#L16 and https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/providers/ytt/02_addons/cpi/cpi_secret.yaml#L6

Reply:
Fixed